cmdstanpy/model.py

```diff
@@ -898,9 +906,28 @@ def sample(
                 method_args=sampler_args,
                 refresh=refresh,
             )
-            runset = RunSet(
-                args=args, chains=chains, chain_ids=chain_ids, time_fmt=time_fmt
-            )  # bookkeeping object for command, result, filepaths
+
+            # 2.28 and up default: if exe compiled w/ threads, run single proc
+            # CmdStanArgs ctor validates model exe file
+            num_threads = threads_per_chain
+            one_process_per_chain = True
+            if (
+                force_one_process_per_chain is None
+                or force_one_process_per_chain is False
```

Comment:
This logic isn't what we discussed in #436 - if `force_one_process_per_chain` is false we should assume this works and not do the check

Reply:
we are trying to protect the user from the unfriendliness of the CmdStan CLI.  if we assume this works and spawn a process with arg `num_chains` present, the error is not helpful at all.

Reply:
If you would like to not allow the user to force the behavior, there is no point having the default value be `None` - you can make the type non-optional and default to false. 

However, `num_chains` as an argument does have behavior that exists even if `STAN_THREADS` was not set. Without an override, there is no way to trigger this in cmdstanpy. Now, would anyone ever want it? Maybe in testing, but probably not in practice.

I think my preferred solution is to say `None` works as it currently does and does the check, `True` does the prior behavior where we manually spawn multiple processes, and `False` always uses 'num_threads', but either outputs a warning or verifies the _version_, but not `STAN_THREADS`. Does that make sense?

Reply:
yes, makes sense.  the version check is necessary.

Reply:
Great. IMO, if `force_one_process_per_chain` is set to `False` by the user, and the version is  < 2.28, that should be an exception, not a 'silently do something else' situation. If they're specifically asking for the behavior, we should do that or throw an error saying we can't